src/config/secret_volatile.rs

```diff
@@ -125,6 +125,39 @@ impl ContactsStore for SecretVolatileConfigStore {
     }
 }
 
+impl MessageStore for SecretVolatileConfigStore {
+    fn save_message(&mut self, _message: libsignal_service::prelude::Content) -> Result<(), Error> {
+        warn!("messages are not saved when using volatile storage.");
+        Ok(())
+    }
+
+    fn messages(&self) -> Result<Vec<libsignal_service::prelude::Content>, Error> {
```

Comment:
I think it would be better if this was `impl Iter<Item = Content>` instead.

Reply:
This is harder then I imagined. Just swapping out `Vec<_>` to `impl Iterator<Item = _>` results in `error[E0562]: `impl Trait` only allowed in function and inherent method return types, not in trait method return`, creating a new type for the returned value works, but one cannot give a concrete value to the type in the implementations as it uses closures for `filter_map`ping. Maybe `Box<dyn Iterator<Item = _>>` is better here?